src/ORM/DataList.php

```diff
@@ -1305,15 +1320,16 @@ public function eagerLoad(...$relations): static
                 $message = "Eager loading only supports up to 3 levels of nesting, passed $count levels - $relation";
                 throw new InvalidArgumentException($message);
             }
-            for ($i = 0; $i < $count; $i++) {
-                if ($i === 0) {
-                    $arr[] = $parts[$i];
-                } else {
-                    $arr[] = $arr[count($arr) - 1] . '.' . $parts[$i];
-                }
+            // Add each relation in the chain as its own entry to be eagerloaded
+            // e.g. for "Players.Teams.Coaches" you'll have three entries:
+            // "Players", "Players.Teams", and "Players.Teams.Coaches
+            $usedParts = [];
+            foreach ($parts as $part) {
+                $usedParts[] = $part;
+                $arr[] = implode('.', $usedParts);
             }
         }
-        $this->eagerLoadRelations = array_merge($this->eagerLoadRelations, $arr);
+        $this->eagerLoadRelations = array_unique(array_merge($this->eagerLoadRelations, $arr));
```

Comment:
Without this change, we'd end up processing the same set of relations twice.
For example, you added
```php
SomeClass::get()->eagerLoad(
    'MyManyMany.MyHasMany',
    'MyManyMany.MySecondHasMany',
);
```
you would have fetched `MyManyMany` for both relation chains.